### PR TITLE
sys-libs/compiler-rt: Provide `atomic-builtins` IUSE flag

### PR DESCRIPTION
**Pull request by vadorovsky**
Atomic builtins fail to build with GCC. Furthermore, they are not needed on systems with GCC, because libatomic can be used on them. Atomic builtins are mostly useful on LLVM-based systems.

Guard them behind the `atomic-builtins` IUSE flag. That flag depends on `clang`.

Bug: https://bugs.gentoo.org/943129

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.
  - **Some checks fail, but not in direct relation to the changes in this PR**

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: sys-libs/compiler-rt/compiler-rt-17.0.6-r2.ebuild [hunk 1]
```diff
@@ -117,6 +116,12 @@ src_configure() {
 		-DPython3_EXECUTABLE="${PYTHON}"
 	)
 
+	if use atomic-builtins ; then
+		mycmakeargs+=(
+			-DCOMPILER_RT_EXCLUDE_ATOMIC_BUILTIN=OFF
```
Comment (thesamesam) [inline sys-libs/compiler-rt/compiler-rt-17.0.6-r2.ebuild:121]: You can just add it to the array, with no explicit conditional, as:
```suggestion
			-DCOMPILER_RT_EXCLUDE_ATOMIC_BUILTIN=$(usex !atomic-builtins)
```

Comment (vadorovsky) [reply sys-libs/compiler-rt/compiler-rt-17.0.6-r2.ebuild]: Fixed, thanks!

File: sys-libs/compiler-rt/compiler-rt-15.0.7-r2.ebuild [hunk 2]
```diff
@@ -12,7 +12,8 @@ HOMEPAGE="https://llvm.org/"
 LICENSE="Apache-2.0-with-LLVM-exceptions || ( UoI-NCSA MIT )"
 SLOT="${LLVM_VERSION}"
 KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~riscv ~x86 ~amd64-linux ~ppc-macos ~x64-macos"
-IUSE="+abi_x86_32 abi_x86_64 +clang debug test"
+IUSE="+abi_x86_32 abi_x86_64 atomic-builtins +clang debug test"
```
Comment (mgorny) [inline sys-libs/compiler-rt/compiler-rt-15.0.7-r2.ebuild:15]:
```suggestion
IUSE="+abi_x86_32 abi_x86_64 +atomic-builtins +clang debug test"
```

Let's default it on?

Comment (vadorovsky) [reply sys-libs/compiler-rt/compiler-rt-15.0.7-r2.ebuild]: Done

**Comment by mgorny** [comment]
Thanks. I'll fix 17, add description to `metadata.xml` and push.

**Pull request closed without merging**